collectors/python.d.plugin/energid/energid.chart.py

```diff
@@ -0,0 +1,132 @@
+        'options': [None, 'Blockchain Difficulty', 'difficulty', 'blockchain', 'energi.difficulty', 'line'],
+        'lines': [
+            ['blockchain.difficulty', 'Diff', 'absolute'],
+        ],
+    },
+    'mempool': {
+        'options': [None, 'MemPool', 'MiB', 'mempool', 'energi.mempool', 'area'],
+        'lines': [
+            ['mempool.max', 'Max', 'absolute', None, 1024*1024],
+            ['mempool.current', 'Usage', 'absolute', None, 1024*1024],
+            ['mempool.txsize', 'TX Size', 'absolute', None, 1024*1024],
+        ],
+    },
+    'secmem': {
+        'options': [None, 'Secure Memory', 'KiB', 'memory', 'energi.secmem', 'area'],
+        'lines': [
+            ['secmem.total', 'Total', 'absolute', None, 1024],
+            ['secmem.locked', 'Locked', 'absolute', None, 1024],
+            ['secmem.used', 'Used', 'absolute', None, 1024],
+        ],
+    },
+    'network': {
+        'options': [None, 'Network', 'count', 'network', 'energi.network', 'line'],
+        'lines': [
+            ['network.connections', 'Connections', 'absolute'],
+        ],
+    },
+    'utxo': {
+        'options': [None, 'UTXO', 'count', 'UTXO', 'energi.utxo', 'line'],
+        'lines': [
+            ['utxo.count', 'UTXO', 'absolute'],
+            ['utxo.xfers', 'Xfers', 'absolute'],
+        ],
+    },
+}
+
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.host = self.configuration.get('host', '127.0.0.1')
+        self.port = self.configuration.get('port', 9796)
+        self.url = 'http://{host}:{port}'.format(
+            host=self.host,
+            port=self.port,
+        )
+        self.method = 'POST'
+        self.header = {
+            'Content-Type': 'application/json',
+        }
+        self.methods = {
+            'getblockchaininfo': lambda r: {
+                'blockchain.blocks': r['blocks'],
+                'blockchain.headers': r['headers'],
+                'blockchain.difficulty': r['difficulty'],
+            },
+            'getmempoolinfo': lambda r: {
+                'mempool.txcount': r['size'],
+                'mempool.txsize': r['bytes'],
+                'mempool.current': r['usage'],
+                'mempool.max': r['maxmempool'],
+            },
+            'getmemoryinfo': lambda r: dict([
+                ('secmem.' + k, v) for (k,v) in r['locked'].items()
+            ]),
+            'getnetworkinfo': lambda r: {
+                'network.timeoffset' : r['timeoffset'],
+                'network.connections': r['connections'],
+            },
+            'gettxoutsetinfo': lambda r: {
+                'utxo.count' : r['txouts'],
+                'utxo.xfers' : r['transactions'],
+                'utxo.size' : r['disk_size'],
+                'utxo.amount' : r['total_amount'],
+            },
+        }
```

Comment:
i think better to move these `self.methods` to global vars, we don't need them to be unique for every instance of Service.